### PR TITLE
Install improvements

### PR DESCRIPTION
**Pull request by Itxaka**
- Dont panic if there is no cc.Install
 - Let the config methods be the ones to print the full config on debug
 - Use vfs utils to create tempfiles

### CODE REVIEW

File: internal/agent/install.go [hunk 1]
```diff
@@ -89,20 +87,24 @@ func ManualInstall(c string, options map[string]string, strictValidations, debug
 	mergeOption(configStr, options)
 
 	if options["device"] == "" {
-		if cc.Install.Device == "" {
-			options["device"] = detectDevice()
-		} else {
-			options["device"] = cc.Install.Device
+		// IF cc is nil we will panic, check beforehand.
+		// If we set no device then we will exit later down the line anyway
+		if cc.Install != nil {
```
Comment (jimmykarily) [inline internal/agent/install.go:92]: Merge the 2 `if`s  with `&&` ?

File: internal/agent/install.go [hunk 2]
```diff
@@ -89,20 +87,24 @@ func ManualInstall(c string, options map[string]string, strictValidations, debug
...
+			if cc.Install.Device == "" {
+				options["device"] = detectDevice()
+			} else {
+				options["device"] = cc.Install.Device
+			}
 		}
 	}
 
+	// Set debug from here already, so it's loaded by the ReadConfigRun
+	viper.Set("debug", debug)
```
Comment (jimmykarily) [inline internal/agent/install.go:102]: If we are ok with sharing the value of `debug` using viper (like we do for `ReadConfigRun`), why don't we set this at the top level (in `main.go`, manual-install action) so we don't need to pass it as an argument to `ManualInstall`?

File: internal/agent/install.go [hunk 3]
```diff
@@ -131,13 +133,13 @@ func Install(debug bool, dir ...string) error {
 
 	ensureDataSourceReady()
 
+	// Set debug from here already, so it's loaded by the ReadConfigRun
+	viper.Set("debug", debug)
```
Comment (jimmykarily) [inline internal/agent/install.go:137]: same

**Review by jimmykarily — APPROVED** [review]

**Comment by Itxaka** [comment]
Fuuuu

This had automerge on 😭

I will implement @jimmykarily feedback into a new PR

**Pull request merged**